src/logic/unique.lean

```diff
@@ -161,3 +161,11 @@ protected def injective.unique [inhabited α] [subsingleton β] (hf : injective
 @unique.mk' _ _ hf.subsingleton
 
 end function
+
+section subtype
+
+instance unique.subtype_eq (y : α) : unique {x // x = y} :=
+{ default := ⟨y, rfl⟩,
+  uniq := λ ⟨x, hx⟩, by simpa using hx }
+
```

Comment:
For completeness, let's add the symmetric version as well. (Untested but works in my head.)
```suggestion

instance unique.subtype_eq' (y : α) : unique {x // y = x} :=
{ default := ⟨y, rfl⟩,
  uniq := λ ⟨x, hx⟩, by simpa using hx }
  
```